pkgs/test_core/lib/src/runner/configuration.dart

```diff
@@ -30,6 +30,16 @@ import 'suite.dart';
 /// The key used to look up [Configuration.current] in a zone.
 final _currentKey = Object();
 
+class PathConfiguration {
+  const PathConfiguration({
+    required this.filePath,
+    this.testName,
+  });
+
+  final String filePath;
+  final Pattern? testName;
```

Comment:
I think we decided that we should allow multiple `name` parameters so this should be a list, maybe just call it `testPatterns`.